src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,75 +2,97 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 
 import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.Currency;
 import java.util.List;
 
 import static org.junit.Assert.*;
 
 public class ArrayListProductDaoTest
 {
-    private ProductDao productDao;
+    @InjectMocks
+    private final ArrayListProductDao productDao = new ArrayListProductDao();
+
+    private List<Product> testProducts;
+
+    @Mock
+    private Product product1 = new Product();
+
+    @Mock
+    private Product product2 = new Product();
+
+    @Mock
+    private Product product3 = new Product();
+
+    @Mock
+    private Product productToSave = new Product();
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        testProducts = new ArrayList<>();
+        Currency usd = Currency.getInstance("USD");
+        product1.setId(15L);
```

Comment:
When working with mocks (so, after you fix the issue above), you setup them using when().thenReturn() methods, setters won't work for them.